From 21ed9a08ac1991fafc516c45bd6732ca13bc122d Mon Sep 17 00:00:00 2001
From: Jason Lowe <jlowe@apache.org>
Date: Fri, 30 Oct 2015 15:31:38 +0000
Subject: [PATCH 1137/2848] MAPREDUCE-6528. Memory leak for
 HistoryFileManager.getJobSummary(). Contributed
 by Junping Du (cherry picked from commit
 6344b6a7694c70f296392b6462dba452ff762109)

(cherry picked from commit a7d8895b60303831a6f65bcc6b6d6bd5b480a1bf)

Change-Id: I604e6cedcde7a7985dc09364d54138afae4cec82
---
 .../hadoop/mapreduce/v2/hs/HistoryFileManager.java |   13 ++++++++++---
 1 file changed, 10 insertions(+), 3 deletions(-)

diff --git a/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-hs/src/main/java/org/apache/hadoop/mapreduce/v2/hs/HistoryFileManager.java b/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-hs/src/main/java/org/apache/hadoop/mapreduce/v2/hs/HistoryFileManager.java
index 83f2700..e827d93 100644
--- a/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-hs/src/main/java/org/apache/hadoop/mapreduce/v2/hs/HistoryFileManager.java
+++ b/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-hs/src/main/java/org/apache/hadoop/mapreduce/v2/hs/HistoryFileManager.java
@@ -997,9 +997,16 @@ private void moveToDoneNow(final Path src, final Path target)
 
   private String getJobSummary(FileContext fc, Path path) throws IOException {
     Path qPath = fc.makeQualified(path);
-    FSDataInputStream in = fc.open(qPath);
-    String jobSummaryString = in.readUTF();
-    in.close();
+    FSDataInputStream in = null;
+    String jobSummaryString = null;
+    try {
+      in = fc.open(qPath);
+      jobSummaryString = in.readUTF();
+    } finally {
+      if (in != null) {
+        in.close();
+      }
+    }
     return jobSummaryString;
   }
 
-- 
1.7.9.5

